updated contributor guide (#1075)
newrelic/docs-website: CONTRIBUTING.md

5. Make a new branch: `git checkout -b your_branch_name`.
6. Pull in content from you private repo: `git pull private main`.
7. Push your new work up to GitHub: `git push origin your_branch_name`.

## Troubleshooting

This section attempts to capture some of the more common troubleshooting techniques to try
if you face issues in contributing.

### General Install, compile or build issues

Many of the common errors you'll face can be resolve by the following:

> 1. Ensure you are using `Yarn` not `NPM`

Reviewer: ```suggestion
1. Ensure you are using `yarn` not `npm`
```

Maybe this is a personal choice, but generally I like referring to these as you'd type them in your terminal. Feel free to ignore this if you don't like this suggestion.

> 2. When in doubt completely delete the `node_module` folder in the root project directory, and re-run `yarn install`

Reviewer: ```suggestion
2. When in doubt completely delete the `node_modules` folder in the root project directory, and re-run `yarn`
```

You may also consider calling this out as a "last resort". This suggestion is the most time consuming, and usually we shouldn't have to do this. It seems every now and then when we upgrade Gatsby, we need to clear out node modules and reinstall all of them to fix some things, but again, this should be a last resort option and not something we need to do often.

By making this 2nd in the list, I worry this might communicate we'd expect you to do this often.

Author: good point

3. Always run `git pull` whenever you intend to create a new working branch

> 4. Always run `Yarn install` whenever you git pull.

Reviewer: ```suggestion
4. Always run `yarn` whenever you git pull.
```

Simplified the install command

Author: got it

### Build / compile issues on the Apple M1 chipset

> 

Reviewer: You might add just a tiny paragraph that says something like:

> If you are working on an M1 Mac, you may run into an error like the following when installing `node_modules`:

Author: added.

```bash

> gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
>   CC(target) Release/obj.target/nothing/../node-addon-api/nothing.o
>   LIBTOOL-STATIC Release/nothing.a
> warning: /Library/Developer/CommandLineTools/usr/bin/libtool: archive library: Release/nothing.a the table of contents is empty (no object file members in the library define global symbols)
>   TOUCH Release/obj.target/libvips-cpp.stamp
>   CXX(target) Release/obj.target/sharp/src/common.o
> ../src/common.cc:23:10: fatal error: 'vips/vips8' file not found
> include <vips/vips8>
> 1 error generated.
> make: *** [Release/obj.target/sharp/src/common.o] Error 1
> gyp ERR! build error
> gyp ERR! stack Error: `make` failed with exit code: 2
> gyp ERR! stack     at ChildProcess.onExit (/Users/cirne/.nvm/versions/node/v15.4.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
> gyp ERR! stack     at ChildProcess.emit (node:events:376:20)
> gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:284:12)
> gyp ERR! System Darwin 20.3.0
> gyp ERR! command "/Users/doe/.nvm/versions/node/v15.4.0/bin/node" "/Users/cirne/.nvm/versions/node/v15.4.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
> gyp ERR! cwd /Users/jcirne/dev/docs-website/node_modules/sharp

Reviewer: ```suggestion
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
  CC(target) Release/obj.target/nothing/../node-addon-api/nothing.o
  LIBTOOL-STATIC Release/nothing.a
warning: /Library/Developer/CommandLineTools/usr/bin/libtool: archive library: Release/nothing.a the table of contents is empty (no object file members in the library define global symbols)
  TOUCH Release/obj.target/libvips-cpp.stamp
  CXX(target) Release/obj.target/sharp/src/common.o
../src/common.cc:23:10: fatal error: 'vips/vips8' file not found
include <vips/vips8>
1 error generated.
make: *** [Release/obj.target/sharp/src/common.o] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/me/.nvm/versions/node/v15.4.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (node:events:376:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:284:12)
gyp ERR! System Darwin 20.3.0
gyp ERR! command "/Users/me/.nvm/versions/node/v15.4.0/bin/node" "/Users/me/.nvm/versions/node/v15.4.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/me/dev/docs-website/node_modules/sharp
```

Generalized the user in the path

```

**Solution**

https://github.com/lovell/sharp/issues/2460#issuecomment-768558230

### Problems with yarn start

> 

Reviewer: ```suggestion
When you start up the site via `yarn start`, you may see errors in your terminal like the following:
```

You might add paragraph like the above to better introduce when you might see the issue.

```bash
 ERROR
[BABEL] Note: The code generator has deoptimised the styling of /Users/jdoe/code/docs-website/src/content/docs/licenses/license-information/other-licenses/services-licenses.mdx as it exceeds the max of 500KB.
```

**Solution**

> ignore it, not a big issue 😉

Reviewer: ```suggestion
You can ignore the error as this is a babel warning when compiling MDX docs. This does not affect the functionality of the site. 😉
```

Added a bit more context on why you can ignore it.
